assets/src/block-editor/components/with-media-library-notice.js

```diff
@@ -37,146 +26,149 @@ export default (InitialMediaUpload, minImageDimensions) => {
+
+				// After bringing in the frame, load the actual editor via an ajax call.
+				view.loadEditor();
+			},
+
+			/**
+			 * Create the default states.
+			 *
+			 * @return {void}
+			 */
+			createStates: function createStates() {
+				this.on(
+					'toolbar:create:featured-image',
+					this.featuredImageToolbar,
+					this
+				);
+				this.on('content:render:edit-image', this.editState, this);
+
+				const FeaturedImageLibrary =
+					wp.media.controller.FeaturedImage.extend({
+						defaults: {
+							...wp.media.controller.FeaturedImage.prototype
+								.defaults,
+							date: false,
+							filterable: false,
+							// Note: These suggestions are shown in the media library image browser.
+							suggestedWidth: EXPECTED_WIDTH,
+							suggestedHeight: EXPECTED_HEIGHT,
+						},
 					});
-				},
-				this.frame
-			);
 
-			this.frame.on('open', this.onOpen);
-
-			this.frame
-				.state('featured-image')
-				.on('select', this.onSelectImage, this);
+				this.states.add([
+					new FeaturedImageLibrary(),
+					new wp.media.controller.EditImage({
+						model: this.options.editImage,
+					}),
+				]);
+			},
+		});
+	};
 
-			// See wp.media() for this.
-			wp.media.frame = this.frame;
-		}
+	/**
+	 * Get attachment collection.
+	 *
+	 * @param {Array} ids Attachment IDs.
+	 * @return {Object}     The attachment collection.
+	 */
+	const getAttachmentsCollection = (ids) => {
+		const { wp } = window;
+
+		return wp.media.query({
+			order: 'ASC',
+			orderby: 'post__in',
+			post__in: ids,
+			posts_per_page: -1,
+			query: true,
+			type: 'image',
+		});
+	};
 
+	/**
+	 * Extends the MediaUpload component to display a notice for small images.
+	 */
+	return class FeaturedImageMediaUpload extends InitialMediaUpload {
 		/**
-		 * Ensure the selected image is the first item in the collection.
+		 * Initializes the Media Library requirements for the featured image flow.
 		 *
-		 * @see https://github.com/WordPress/gutenberg/blob/c58b32266f8c950c5b9927d286608343078aee02/packages/media-utils/src/components/media-upload/index.js#L401-L417
-		 */
-		onOpen() {
-			const frameContent = this.frame.content.get();
-			if (frameContent && frameContent.collection) {
-				const collection = frameContent.collection;
-
-				// Clean all attachments we have in memory.
-				collection
-					.toArray()
-					.forEach((model) => model.trigger('destroy', model));
-
-				// Reset has more flag, if library had small amount of items all items may have been loaded before.
-				collection.mirroring._hasMore = true;
-
-				// Request items.
-				collection.more();
-			}
-		}
-
-		/**
-		 * Handles image selection.
+		 * @override
+		 * @description Overrides the media upload component's initialize method for featured image.
+		 * @see {@link https://github.com/WordPress/gutenberg/blob/debddee2ace15263c08c66b5f5a43a9e17bf5d0c/packages/media-utils/src/components/media-upload/index.js#L301} Original MediaUpload buildAndSetFeatureImageFrame method.
```

Comment:
For some reason I'm getting a syntax problem here in PhpStorm:

![image](https://github.com/ampproject/amp-wp/assets/134745/6137c102-95c7-469e-9f64-81e99c70773e)

This fixes it:

```suggestion
		 * Modified from {@link https://github.com/WordPress/gutenberg/blob/debddee2ace15263c08c66b5f5a43a9e17bf5d0c/packages/media-utils/src/components/media-upload/index.js#L301-L316|Original MediaUpload buildAndSetFeatureImageFrame method}.
```